ext/npm-extension.js

```diff
@@ -258,6 +258,11 @@ exports.addExtension = function(System){
 	System.locate = function(load){
 		var parsedModuleName = utils.moduleName.parse(load.name),
 			loader = this;
+
+		if (load.metadata) {
+			load.metadata.parsedModuleName = parsedModuleName;
```

Comment:
Arggghh, I totally missed that the test is calling `locate` straight passing a partial load object 

```js
```